src/C1T2X_OBU.py

```diff
@@ -114,54 +126,61 @@ def sendLAN(lPacket):
 def VANET_listening_thread():
 	global error
 
-	while not error and not vanet.error:
-
-		try:
-			pkt = vanet.recv_packets()
-			c1t2x_logger.debug("Received %s from VANET" %pkt)
-			if pkt:
-				if not parseVANETPacket:
-					sendLAN(pkt[0])
-				else:
-					# feature to parse incoming VANET message is not yet enabled
-					c1t2x_logger.error("Feature to parse incoming VANET message is not yet enabled")
-					raise NotImplementedError
+	while not vanet.error:
+		with mutex:
+			if error:
+				break
+
+			try:
+				pkt = vanet.recv_packets()
+				c1t2x_logger.debug("Received %s from VANET", pkt)
+				if pkt:
+					if not parseVANETPacket:
+						sendLAN(pkt[0])
+					else:
+						# feature to parse incoming VANET message is not yet enabled
+						c1t2x_logger.error("Feature to parse incoming VANET message is not yet enabled")
+						raise NotImplementedError
+					if printData:
+						print(pkt)
+			except:
 				if printData:
-					print(pkt)
-		except:
-			if printData:
-				print("Waiting to configure LAN")
-			c1t2x_logger.info("Waiting to configure LAN")
-			time.sleep(0.25)
-		time.sleep(loopTime)
+					print("Waiting to configure LAN")
+				c1t2x_logger.info("Waiting to configure LAN")
+				time.sleep(0.25)
+			time.sleep(loopTime)
 
 	error = True
 	c1t2x_logger.info("Terminating VANET Thread")
 
 def LAN_listening_thread():
 	global error
 
-	while not error and not lan.error:
-		try:
-			pkt = lan.recv_packets()
-			if pkt:
-				if not parseLANPacket:
-					sendVANET(pkt[0])
-				else:
-					# feature to parse incoming LAN packet is not enabled
-					# this feature may be used for things like responding to requests from the LAN connection, etc.
-					# there is no intent for this feature to be enabled but it is being allocated space in the code
-					c1t2x_logger.error("Feature to parse incoming LAN is not enabled")
-					raise NotImplementedError
-		except:
-			if printData:
-				print("Waiting to configure VANET")
-			c1t2x_logger.debug("Waiting to configure VANET")
-			time.sleep(0.25)
-		time.sleep(loopTime)
+	while not lan.error:
+		with mutex:
+			if error:
+				c1t2x_logger.info("Terminating LAN Thread")
+				break
+
+			try:
+				pkt = lan.recv_packets()
+				if pkt:
+					if not parseLANPacket:
+						sendVANET(pkt[0])
+					else:
+						# feature to parse incoming LAN packet is not enabled
+						# this feature may be used for things like responding to requests from the LAN connection, etc.
+						# there is no intent for this feature to be enabled but it is being allocated space in the code
+						c1t2x_logger.error("Feature to parse incoming LAN is not enabled")
+						raise NotImplementedError
+			except:
+				if printData:
+					print("Waiting to configure VANET")
+				c1t2x_logger.debug("Waiting to configure VANET")
+				time.sleep(0.25)
+			time.sleep(loopTime)
```

Comment:
This does not need to be within the mutex's scope. Same applies to other thread.